grammar.md

```diff
@@ -41,12 +41,12 @@ ReturnStatement                     return Expression? ;
 Expression                          LogicalOrExpression (= LogicalOrExpression)*                           $ Self.out = ast::Expr::Assignment(LogicalOrExpression_1, LogicalOrExpression_2.all_outs)
 
 // BEGIN precedence climbing
-LogicalOrExpression                 LogicalAndExpression (\|\| LogicalAndExpression)*                      $ Self.out = ast::Expr::Binary(<binop>, Lhs, Rhs)
-LogicalAndExpression                EqualityExpression (&& EqualityExpression)*                            $ <analog>
-EqualityExpression                  RelationalExpression ( (== | !=) RelationalExpression)*                $ <analog>
-RelationalExpression                AddititveExpression ( (< | <= | > | >=) AdditiveExpression )*          $ <analog>
-AdditiveExpression                  MultiplicativeExpression ( (+|-) MultiplicativeExpression )*           $ <analog>
-MultiplicativeExpression            UnaryExpression ( (\*|/|%) UnaryExpression)*                           $ <analog>
+LogicalOrExpression                 (LogicalAndExpression \|\|)? LogicalAndExpression                      $ <analog>
```

Comment:
Oh I missed that on my review: shouldn't this be `*` and not `?`

Reply:
I'm not sure. I wanted to revert the grammer to a state that reflechts the associativity correctly. I don't think thats achievable via `*`. But the recursion is broken, your're right. It should probably be:
```
LogicalOrExpression                 (LogicalOrExpression \|\|)? LogicalAndExpression
```
Doesn't really matter, since this is implemented with a precedence parser anyway, not in terms of the grammer